src/pages/Swap/LimitOrderBox/api/cow.ts

```diff
@@ -33,23 +33,41 @@ type GetLimitOrderQuoteParams = SignLimitOrderParams
  * @returns
  */
 export async function getQuote({ order, signer, chainId }: GetLimitOrderQuoteParams) {
-  const { buyToken, receiverAddress, userAddress, expiresAt, sellAmount, sellToken, kind } = order
+  const { buyToken, receiverAddress, userAddress, expiresAt, sellAmount, sellToken, kind, buyAmount } = order
 
   // const cowSdk = getCoWSdk(chainId, signer)
   const cowSdk = CoWTrade.getCowSdk(chainId, {
     signer,
     appDataHash: getAppDataIPFSHash(chainId),
   })
 
-  const cowQuote = await cowSdk.cowApi.getQuote({
-    buyToken,
-    sellToken,
-    amount: sellAmount,
-    validTo: expiresAt,
-    kind: kind === LimitOrderKind.BUY ? CoWOrderKind.BUY : CoWOrderKind.SELL,
-    receiver: receiverAddress,
-    userAddress: userAddress,
-  })
+  let cowQuote: Awaited<ReturnType<typeof cowSdk.cowApi.getQuote>>
+
+  if (kind === LimitOrderKind.BUY) {
+    cowQuote = await cowSdk.cowApi.getQuote({
+      appData: getAppDataIPFSHash(chainId),
+      buyAmountAfterFee: buyAmount,
+      buyToken,
+      from: userAddress,
+      kind: CoWOrderKind.BUY,
+      partiallyFillable: false,
+      receiver: receiverAddress,
+      sellToken,
+      validTo: expiresAt,
+    })
+  } else {
+    cowQuote = await cowSdk.cowApi.getQuote({
+      appData: getAppDataIPFSHash(chainId),
+      buyToken,
+      sellAmountBeforeFee: sellAmount,
+      from: userAddress,
+      kind: CoWOrderKind.SELL,
+      partiallyFillable: false,
+      receiver: receiverAddress,
+      sellToken,
+      validTo: expiresAt,
+    })
+  }
```

Comment:
@adamazad I guess I should have expected code sabotage when I saw your name on the commit :D
![Screenshot 2022-12-27 at 17 17 58](https://user-images.githubusercontent.com/33226956/209693023-5b5bf24a-febe-485b-8879-bc8abf6ef8ee.png)

Reply:
I explained on the standup. See https://github.com/cowprotocol/cow-sdk/issues/91